src/debugpy/server/cli.py

```diff
@@ -304,15 +316,18 @@ def start_debugging(argv_0):
 
     debugpy.configure(options.config)
 
-    if options.mode == "listen" and options.address is not None:
-        debugpy.listen(options.address)
-    elif options.mode == "connect" and options.address is not None:
-        debugpy.connect(options.address, access_token=options.adapter_access_token)
-    else:
-        raise AssertionError(repr(options.mode))
+    if os.environ.get("DEBUGPY_RUNNING", "false") != "true":
```

Comment:
Okay so this is specifically the case that a subprocess shouldn't connect. I'm wondering if there's a situation that this might break. Like if you wanted to connect to subprocesses too. I believe that's what the [subProcess](https://github.com/microsoft/debugpy/blob/fb6158a059bcc6a0391eb0f670d5885d664c102d/tests/debug/config.py#L47) flag is for. Would this change prevent all `subProcess` connections from working?

Could you instead pass `--config-subprocess=false` during your initial start of flask? (as described here: https://github.com/microsoft/debugpy/blob/fb6158a059bcc6a0391eb0f670d5885d664c102d/README.md?plain=1#L61)

Reply:
If that doesn't work, I don't think this is the fix we'd want. It seems like the reload from werkzeug should cause the original process to die before reloading. Maybe we can do that here (wait for the original server to finish if we get an address in use).

Reply:
Ok so I'm trying to verify this use case but I can't get it working. I'm using the example app

```python
from flask import Flask
from multiprocessing import Process, Queue
from time import sleep

app = Flask(__name__)


def long_calculation(queue: Queue):
    sleep(5)
    queue.put(5)


@app.route("/")
def get_data():
    queue = Queue()
    p = Process(target=long_calculation, args=(queue,))
    p.start()
    p.join()
    res = queue.get()
    return str(res), 200

```

which I'm running with the command
```
python -m debugpy --listen 127.0.0.1:5678 --configure-subProcess False -m flask -A playground run --no-debug
```

However, if I set a breakpoint in the helper function in VS Code and connect, it blocks at the breakpoint without showing this in the UI, so that the only way to unblock things is to shut down the program. 

If I'm doing something wrong here, can you indicate how the flag is supposed to work? It's difficult to test if my fix would break this functionality without knowing how it's supposed to work in the first place :sweat_smile:

Reply:
Not sure what you mean it blocks at the breakpoint in the UI. Wouldn't you expect it to hit the breakpoint?

That flag should prevent automatic attaching to child processes.

I can't reproduce the original issue myself. Debugging works fine with the latest werkzeug. Reload works too.

Reply:
If it doesn't attach to child processes, I wouldn't expect it to hit a breakpoint inside a function running in such a process, but it does.

Reply:
Okay I see what you mean about blocking the UI. If I stick a breakpoint in the subprocess code (and subProcess is false), the request never finishes. Without the breakpoint it works fine.

Seems like a separate issue, but yes it doesn't debug a child process if you pass the `--configure-subProcess False`. Which seems like what you'd want. I'm afraid your change would cause the same thing for all sub processes.

So with your change, can you still hit a breakpoint in the `long_calculation`?

Reply:
I just tried it and it seems like it works still. It doesn't force subprocess debugging off.

Reply:
Yes I can

Reply:
It still works because subprocesses don't use this code path to connect. Rather, pydevd itself handles subprocesses [here](src/debugpy/_vendored/pydevd/_pydev_bundle/pydev_monkey.py), such that the subprocess immediately connects to the debugpy adapter process (which then propagates this new connection to the client).